File: ios/SwiftKotlination/SwiftKotlinationTests/Mocks/Features/TopStories/Model/TopStoriesRepositoryMock.swift
import XCTest
import RxSwift
@testable import SwiftKotlination

final class TopStoriesRepositoryMock: TopStoriesRepositoryProtocol {
    
    var storiesStub: StoriesStub
    
    init(storiesStub: StoriesStub = .failure(.timeout)) {
        self.storiesStub = storiesStub
    }
    
    enum StoriesStub {
        case success([Story])
        case failure(RxError)
    }
    
    var stories: Observable<[Story]> {
        switch storiesStub {
        case .success(let stories):
            return Observable.just(stories)
        case .failure(let error):
            return Observable.error(error)
        }
    }
    
}

File: android/SwiftKotlination/app/src/test/java/fr/jhandguy/swiftkotlination/features/topstories/viewmodel/TopStoriesRepositoryMock.kt
package fr.jhandguy.swiftkotlination.features.topstories.viewmodel

import fr.jhandguy.swiftkotlination.features.story.model.Story
import fr.jhandguy.swiftkotlination.features.topstories.model.TopStoriesRepository
import io.reactivex.Observable

class TopStoriesRepositoryMock(stories: List<Story> = ArrayList(), error: Error? = null): TopStoriesRepository {

    override var topStories: Observable<List<Story>> = when {
        error != null -> Observable.error(error)
        else -> Observable.just(stories)
    }
}